protocol/messenger_backup_handler.go

```diff
@@ -0,0 +1,205 @@
+package protocol
+
+import (
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/multiaccounts/settings"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/wakusync"
+)
+
+const (
+	// timeToPostponeBackedUpMessagesHandling - the idea is to wait for 30 secs in loading state with the hope that at least
+	// one message from the set of last backed up messages will arrive, that we can know which set of messages to work with
+	timeToPostponeBackedUpMessagesHandling = 30 * time.Second
+
+	SyncWakuSectionKeyProfile     = "profile"
+	SyncWakuSectionKeyContacts    = "contacts"
+	SyncWakuSectionKeyCommunities = "communities"
+	SyncWakuSectionKeySettings    = "settings"
+)
+
+type backupHandler struct {
+	messagesToProceed      []protobuf.Backup
+	lastKnownTime          uint64
+	postponeHandling       bool
+	postponeTasksWaitGroup sync.WaitGroup
+}
+
+func (bh *backupHandler) addMessage(message protobuf.Backup) {
+	if message.Clock < bh.lastKnownTime {
+		return
+	}
+	if message.Clock > bh.lastKnownTime {
+		bh.messagesToProceed = nil
+		bh.lastKnownTime = message.Clock
+	}
+
+	bh.messagesToProceed = append(bh.messagesToProceed, message)
+}
+
+func (m *Messenger) startWaitingForTheLatestBackedupMessageLoop() {
+	ticker := time.NewTicker(timeToPostponeBackedUpMessagesHandling)
+	m.backedupMessagesHandler.postponeTasksWaitGroup.Add(1)
+	go func() {
+		defer m.backedupMessagesHandler.postponeTasksWaitGroup.Done()
+		for {
+			select {
+			case <-ticker.C:
+				if !m.backedupMessagesHandler.postponeHandling {
+					return
```

Comment:
Should it be `return` or `continue`?  the `return` will kill the goroutine.

Reply:
Yes, it should be return, cause that go routine is run only once when user logs in the app following one of two onboarding flows for old Status' user.